pkgs/development/libraries/criterion/default.nix

```diff
@@ -1,26 +1,31 @@
-{ lib, stdenv, fetchFromGitHub, boxfort, cmake, libcsptr, pkg-config, gettext
-, dyncall , nanomsg, python3Packages }:
+{ lib, stdenv, fetchFromGitHub, boxfort, meson, libcsptr, pkg-config, gettext
+, cmake, ninja, protobuf, libffi, libgit2, dyncall, nanomsg, nanopb, python3Packages, git }:
 
-stdenv.mkDerivation rec {
-  version = "2.3.3";
+let
+  nanopbMalloc = nanopb.override { mallocBuild = true; };
+in stdenv.mkDerivation rec {
+  version = "2.4.1";
   pname = "criterion";
 
   src = fetchFromGitHub {
     owner = "Snaipe";
     repo = "Criterion";
     rev = "v${version}";
-    sha256 = "0y1ay8is54k3y82vagdy0jsa3nfkczpvnqfcjm5n9iarayaxaq8p";
+    sha256 = "KT1XvhT9t07/ubsqzrVUp4iKcpVc1Z+saGF4pm2RsgQ=";
     fetchSubmodules = true;
   };
 
-  nativeBuildInputs = [ cmake pkg-config ];
+  nativeBuildInputs = [ meson ninja cmake pkg-config protobuf git ];
```

Comment:
Do we really need meson, ninja, cmake and pkg-config? I imagine that we can drop either ninja or meson. The warnings from meson are informational and can be ignored.

Also why do we need git?

Reply:
This specific build script won't work without CMake, Ninja and Git as builddeps.

Reply:
Then we can remove pkg-config?

Where is git used? Isn't it optional https://github.com/Snaipe/Criterion/blob/354341759c48a15d6396579b99f4a01b7c3ae081/meson.build#L18 ? Can you post a error log?

Reply:
Well, right after that line it blindly calls the `git` command on the basis that the source is a Git repo, so that optional flag is a lie.

Reply:
Can we hardcode the line below that to False? git shouldn't really be in nativeBuildInputs because we normally do not have a .git directory or internet connection to fetch things.

Reply:
well I safely dropped Git after testing. No idea why Meson was complaining before